iroh/src/commands/sync.rs

```diff
@@ -195,62 +186,44 @@ impl DocCommands {
+                let key = key.as_bytes().to_vec();
+                let filter = match (author, prefix) {
+                    (None, false) => GetFilter::Key(key),
+                    (None, true) => GetFilter::Prefix(key),
+                    (Some(author), false) => {
+                        let entry = doc
+                            .get_one(author, key)
+                            .await?
+                            .ok_or_else(|| anyhow!("Entry not found"))?;
+                        print_entry(&doc, &entry, content).await?;
+                        return Ok(());
+                    }
+                    (Some(author), true) => GetFilter::AuthorAndPrefix(author, key),
                 };
 
                 let mut stream = doc.get(filter).await?;
                 while let Some(entry) = stream.try_next().await? {
-                    println!("{}", fmt_entry(&entry));
-                    if content {
-                        if entry.content_len() < MAX_DISPLAY_CONTENT_LEN {
-                            match doc.get_content_bytes(entry.content_hash()).await {
-                                Ok(content) => match String::from_utf8(content.into()) {
-                                    Ok(s) => println!("{s}"),
-                                    Err(_err) => println!("<invalid UTF-8>"),
-                                },
-                                Err(err) => println!("<failed to get content: {err}>"),
-                            }
-                        } else {
-                            println!(
-                                "<skipping content with len {}: too large to print>",
-                                HumanBytes(entry.content_len())
-                            )
-                        }
-                    }
-                    println!();
+                    print_entry(&doc, &entry, content).await?;
                 }
             }
             Self::Keys {
                 doc,
-                old,
                 prefix,
                 author,
             } => {
                 let doc = iroh.get_doc(env.doc(doc)?).await?;
-                let filter = match old {
-                    true => GetFilter::all(),
-                    false => GetFilter::latest(),
-                };
-                let mut filter = match prefix {
-                    Some(prefix) => filter.with_prefix(prefix),
-                    None => filter,
-                };
-                if let Some(author) = author {
-                    filter = filter.with_author(author);
+                let filter = match (author, prefix) {
+                    (None, None) => GetFilter::All,
+                    (None, Some(prefix)) => GetFilter::Prefix(prefix.as_bytes().to_vec()),
+                    (Some(author), None) => GetFilter::Author(author),
+                    (Some(author), Some(prefix)) => {
+                        GetFilter::AuthorAndPrefix(author, prefix.as_bytes().to_vec())
+                    }
```

Comment:
maybe this could be a method on `GetFilter` sth like `From<(Option<AuthorId>, &[u8])> for GetFilter`

Reply:
Made a method instead, not sure if the `From` for a tuple is discoverable enough.